File: programmers/lev1/12943.py
def solution(num):
    answer = 0
    cnt=0
    if num!=1:
        while cnt<500:
            if num%2==0:
                num//=2
                cnt+=1
                if num==1:
                    break
            else:
                num=(num*3)+1
                cnt+=1
                if num==1:
                    break
    else:
        return 0
    if cnt==500:
        answer=-1
    else:
        answer=cnt
    return answer

File: java/src/programmers/year2023/lv1/_12943.java
package programmers.year2023.lv1;

public class _12943 {
    public static void main(String[] args) {
        int num = 626331;
        solution(num);
    }

    public static int solution(int num) {
        int answer = 0;
        long input = num;
        if (input == 1) {
            return 0;
        }
        while (input != 1) {
            answer += 1;
            if (answer == 500) {
                return -1;
            }

            if (input % 2 == 0) {
                input /= 2;
            } else {
                input *= 3;
                input += 1;
            }
        }
        System.out.println("num = " + num);
        System.out.println("answer = " + answer);
        return answer;
    }
}
